reflex/experimental/vars/object.py

```diff
@@ -88,18 +104,19 @@ class LiteralObjectVar(LiteralVar, ObjectVar):
     def __init__(
         self,
         _var_value: dict[Var | Any, Var | Any],
-        _var_type: Type = dict,
+        _var_type: Type | None = None,
         _var_data: VarData | None = None,
     ):
         """Initialize the object var.
 
         Args:
             _var_value: The value of the var.
+            _var_type: The type of the var.
             _var_data: Additional hooks and imports associated with the Var.
         """
         super(LiteralObjectVar, self).__init__(
             _var_name="",
-            _var_type=_var_type,
+            _var_type=type(_var_value) if _var_type is None else _var_type,
```

Comment:
not sure if the `type` cast is needed here, because `type(dict[str, str])` returns `<class 'types.GenericAlias'>` and we almost certainly want the former so we can call `typing.get_args` on it